src/Registry.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+/**
+@title Simple Name Registry
+@author hashedMae
+@notice A simple name registry that stores the data on chain
+ */
+contract Registry {
+
+
+    ///@notice Mapping of a name to address of currentOwner
+     mapping(string => address) public nameToUser;
+
+    ///@dev events for tracking when names are registered or released
+    event NameRegistered(string name, address user);
+    event NameReleased(string name, address user);
+
+    /**
+    @notice checks if a name is currently registered to a user
+    @dev for internal use only, not publicly callable
+    @param _name string that is being checked if it is registered
+    @return true if name is currently registered to a user, otherwise false
+     */
+     function _isRegistered(string memory _name) internal view returns(bool) {
+         if(nameToUser[_name] != address(0x0)){
+             return true;
+         }
+         return false;
+     }
+
+     /**
+     @notice function to register a name
+     @dev emits NameRegistered on success
+     @param name string that is being registered
+      */
+    function registerName(string memory name) public {
```

Comment:
we aren't going to get into gas optimization until issue 3 or 4, but you should lookup [calldata](https://docs.soliditylang.org/en/v0.8.13/types.html?highlight=calldata#data-location)

after you've read up on it, you can get more info on this comment on Raja's pr:
https://github.com/calnix/simple-registry/pull/4#pullrequestreview-951121601